player/command.c

```diff
@@ -2381,6 +2381,34 @@ static int panscan_property_helper(void *ctx, struct m_property *prop,
     return r;
 }
 
+// Properties retrieved through VOCTRL_PERFORMANCE_DATA
+static int perfdata_property_helper(void *ctx, struct m_property *prop,
+                                    int action, void *arg)
+{
+    MPContext *mpctx = ctx;
+    if (!mpctx->video_out)
+        return M_PROPERTY_UNAVAILABLE;
+
+    struct voctrl_performance_data data = {0};
+    if (vo_control(mpctx->video_out, VOCTRL_PERFORMANCE_DATA, &data) <= 0)
+        return M_PROPERTY_UNAVAILABLE;
```

Comment:
Keep in mind that sending voctrls every frame also have performance implications. It forces the main thread to sync with the VO thread, while the code carefully tried to avoid such blocking before. Also, since most time the VO thread is blocked on the GL swapbuffer call (for vsync), the voctrl could be significantly delayed each frame, up to the point where the main thread blocks until the VO has no frame to display anymore, unblocking the main thread for a while.

But that's not my problem fortunately.